python/py-luma.emulator/Portfile

```diff
@@ -0,0 +1,39 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+PortGroup           github 1.0
+PortGroup           python 1.0
+
+name                py-luma.emulator
+github.setup        rm-hull luma.emulator 1.5.0
+revision            0
+categories          python
+maintainers         nomaintainer
+platforms           any
+supported_archs     noarch
+license             MIT
+
+description         Drawing and text rendering for small displays, emulator \
+                    module
+long_description    luma.emulator provides a series of pseudo-display devices \
+                    which allow the luma.core components to be used without \
+                    running a physical device.
+
+homepage            https://github.com/rm-hull/luma.core
+
+checksums           rmd160  30ddc55f750c463b3d757f6cf17869c365c3b475 \
+                    sha256  dc3733c0785e95ecd2d96e2137c15864268c1f684f863c7d97b1b009210b0f3e \
+                    size    903573
+python.versions     38 39 310 311
+python.pep517       yes
+
+if {${name} ne ${subport}} {
+    depends_run-append \
+                    port:py${python.version}-luma.core \
+                    port:py${python.version}-game
+    depends_test-append \
+                    port:py${python.version}-pytest-timeout
+    test.run        yes
+    test.env-append PYTHONPATH=${worksrcpath}/build/lib:${worksrcpath}
+    test.args
```

Comment:
did you try without overriding `test.env` and `test.arg` - these should be set correctly by the `python` PG in 99.99% of the cases

Reply:
Yes both are definitely necessary.

It’s been about 6 weeks since I did it and I’d have to recheck to be positive, but as I recall:

`PYTHONPATH` was necessary in order for the relative imports used during testing to work.

`test.args` was necessary because this module comes with its own `pytest.ini` that specifies `addopts` that are crucial to the tests’ success, and the `python` PortGroup adding `-o addopts=''` prevented those required `addopts` from having their desired effect.

Do you want me to verify my memory and add comments explaining the situation?

Reply:
`PYTHONPATH` needs `${worksrcpath}` in it in order to be able to run tests. The tests, such as [`${worksrcpath}/tests/test_asciiblock.py`, have constructs like](https://github.com/rm-hull/luma.emulator/blob/23a74c0a99f7910a8d3a5f61b95cee8a42df656a/tests/test_asciiblock.py#L19):

```
from .helpers import md5, redirect_stdout
```

These imports intend to load [`${worksrcpath}/tests/helpers.py`](https://github.com/rm-hull/luma.emulator/blob/23a74c0a99f7910a8d3a5f61b95cee8a42df656a/tests/helpers.py). Although the tests _run_ from the `tests` directory, importing from that directory without additional `PYTHONPATH` help is not possible.

I set `PYTHONPATH` to `${worksrcpath}/build/lib:${worksrcpath}`, which includes the existing `${worksrcpath}/build/lib` for access to the built `luma.emulator` module, and `${worksrcpath}` for access to `tests.helpers`.

`test.args` is cleared for [the reason I described yesterday](https://github.com/rm-hull/luma.emulator/blob/23a74c0a99f7910a8d3a5f61b95cee8a42df656a/pytest.ini).